packages/@react-spectrum/s2/src/Table.tsx

```diff
@@ -11,85 +11,1096 @@
+                {children}
+              </ColumnContents>
+            )
+          }
         </>
       )}
-    </AriaColumn>
+    </RACColumn>
   );
 }
 
-export function TableHeader<T extends object>(
-  {columns, children}: TableHeaderProps<T>
-) {
-  let {selectionBehavior, selectionMode, allowsDragging} = useTableOptions();
+const columnContentWrapper = style({
+  minWidth: 0,
+  display: 'flex',
+  alignItems: 'center',
+  width: 'full'
+});
+
+const sortIcon = style({
+  size: fontRelative(16),
+  flexShrink: 0,
+  marginEnd: {
+    default: 8,
+    isButton: 'text-to-visual'
+  },
+  verticalAlign: {
+    default: 'bottom',
+    isButton: 0
+  },
+  '--iconPrimary': {
+    type: 'fill',
+    value: 'currentColor'
+  }
+});
+
+interface ColumnContentProps extends Pick<ColumnRenderProps, 'allowsSorting' | 'sortDirection'>, Pick<ColumnProps, 'children'> {}
+
+function ColumnContents(props: ColumnContentProps) {
+  let {allowsSorting, sortDirection, children} = props;
 
   return (
-    <AriaTableHeader>
-      {/* Add extra columns for drag and drop and selection. */}
-      {allowsDragging && <AriaColumn />}
-      {selectionBehavior === 'toggle' && (
-        <AriaColumn>
-          {selectionMode === 'multiple' && <Checkbox slot="selection" />}
-        </AriaColumn>
+    <div className={columnContentWrapper}>
+      {allowsSorting && (
+        <Provider
+          values={[
+            [IconContext, {
+              styles: sortIcon({})
+            }]
+          ]}>
+          {sortDirection != null && (
+            sortDirection === 'ascending' ? <SortUpArrow /> : <SortDownArrow />
+          )}
+        </Provider>
       )}
-      <Collection items={columns}>
+      <span className={style({truncate: true, width: 'full'})}>
         {children}
-      </Collection>
-    </AriaTableHeader>
+      </span>
+    </div>
   );
 }
 
-export function Row<T extends object>(
-  {id, columns, children, ...otherProps}: RowProps<T>
-) {
-  let {selectionBehavior, allowsDragging} = useTableOptions();
+const resizableMenuButtonWrapper = style({
+  ...cellFocus,
+  color: 'gray-800', // body-color
+  width: 'full',
+  position: 'relative',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: {
+    align: {
+      default: 'start',
+      center: 'center',
+      end: 'end'
+    }
+  },
+  // TODO: when align: end, the dropdown arrow is misaligned with the text, not sure how best to make the svg be flush with the end of the button other than modifying the
+  // paddingEnd
+  paddingX: 16,
+  backgroundColor: 'transparent',
+  borderStyle: 'none',
+  fontSize: 'control',
+  fontFamily: 'sans',
+  fontWeight: 'bold'
+});
+
+const resizerHandleContainer = style({
+  display: {
+    default: 'none',
+    isResizing: 'block',
+    isHovered: 'block'
+  },
+  width: 12,
+  height: 'full',
+  position: 'absolute',
+  top: 0,
+  insetEnd: size(-6),
+  cursor: {
+    default: 'none',
+    resizableDirection: {
```

Comment:
oh thats right, completely slipped my mind, missed it since I didn't tag it with TODO lol. I'll add it to the followup ticket and see if we need to submit something to the icons team

Reply:
filed an airtable ticket for those cursors